### PR TITLE
Add `--profile` option to `llvm-kompile`

### PR DESCRIPTION
**Pull request by Baltoli**
This PR makes some minor refactorings to the `llvm-kompile-*` scripts, and adds an additional profiling option that (in verbose mode) prints the time taken by each step of the compilation pipeline.

I will follow this PR up with additional profiling information in the code generator.

For a small example:
```console
$ llvm-kompile test-kompiled/definition.kore test-kompiled/dt main -v --profile
+ /home/bruce/code/llvm-backend/build/install/bin/llvm-kompile-codegen test-kompiled/definition.kore test-kompiled/dt/dt.yaml test-kompiled/dt 0
  time=0.034
+ /usr/lib/llvm-14/bin/opt -mem2reg -tailcallelim -tailcallopt tmp.gsYQm9W8RC -o tmp.CTS0WVHve7
  time=0.021
+ /home/bruce/code/llvm-backend/build/install/bin/llvm-kompile-clang tmp.CTS0WVHve7 main nolto -fno-stack-protector -v --profile -fuse-ld=lld --python-output-dir test-kompiled
  | + /usr/lib/llvm-14/bin/llc -tailcallopt tmp.CTS0WVHve7 -mtriple=x86_64-pc-linux-gnu -filetype=obj -O0 --relocation-model=pic -o tmp.H3lGmLF9OO/asm.o
  |   time=0.040
  | + /usr/lib/llvm-14/bin/llc -tailcallopt /home/bruce/code/llvm-backend/build/install/bin/../lib/kllvm//llvm/finish_rewriting.ll -mtriple=x86_64-pc-linux-gnu -filetype=obj -O0 --relocation-model=pic -o tmp.H3lGmLF9OO/finish_rewriting.ll.o
  |   time=0.010
  | + /usr/lib/llvm-14/bin/llc -tailcallopt /home/bruce/code/llvm-backend/build/install/bin/../lib/kllvm//llvm/fresh.ll -mtriple=x86_64-pc-linux-gnu -filetype=obj -O0 --relocation-model=pic -o tmp.H3lGmLF9OO/fresh.ll.o
  |   time=0.010
  | + /usr/lib/llvm-14/bin/llc -tailcallopt /home/bruce/code/llvm-backend/build/install/bin/../lib/kllvm//llvm/getTag.ll -mtriple=x86_64-pc-linux-gnu -filetype=obj -O0 --relocation-model=pic -o tmp.H3lGmLF9OO/getTag.ll.o
  |   time=0.009
  | + /usr/lib/llvm-14/bin/llc -tailcallopt /home/bruce/code/llvm-backend/build/install/bin/../lib/kllvm//llvm/move_float.ll -mtriple=x86_64-pc-linux-gnu -filetype=obj -O0 --relocation-model=pic -o tmp.H3lGmLF9OO/move_float.ll.o
  |   time=0.009
  | + /usr/lib/llvm-14/bin/llc -tailcallopt /home/bruce/code/llvm-backend/build/install/bin/../lib/kllvm//llvm/move_int.ll -mtriple=x86_64-pc-linux-gnu -filetype=obj -O0 --relocation-model=pic -o tmp.H3lGmLF9OO/move_int.ll.o
  |   time=0.011
  | + /usr/lib/llvm-14/bin/llc -tailcallopt /home/bruce/code/llvm-backend/build/install/bin/../lib/kllvm//llvm/string_equal.ll -mtriple=x86_64-pc-linux-gnu -filetype=obj -O0 --relocation-model=pic -o tmp.H3lGmLF9OO/string_equal.ll.o
  |   time=0.009
  | + /usr/lib/llvm-14/bin/llc -tailcallopt /home/bruce/code/llvm-backend/build/install/bin/../lib/kllvm//llvm/take_steps.ll -mtriple=x86_64-pc-linux-gnu -filetype=obj -O0 --relocation-model=pic -o tmp.H3lGmLF9OO/take_steps.ll.o
  |   time=0.010
  | + /usr/bin/clang++-14 -Wno-override-module -Wno-return-type-c-linkage tmp.H3lGmLF9OO/asm.o tmp.H3lGmLF9OO/finish_rewriting.ll.o tmp.H3lGmLF9OO/fresh.ll.o tmp.H3lGmLF9OO/getTag.ll.o tmp.H3lGmLF9OO/move_float.ll.o tmp.H3lGmLF9OO/move_int.ll.o tmp.H3lGmLF9OO/string_equal.ll.o tmp.H3lGmLF9OO/take_steps.ll.o /home/bruce/code/llvm-backend/build/install/bin/../lib/kllvm//libarithmetic.a /home/bruce/code/llvm-backend/build/install/bin/../lib/kllvm//llvm/main/main.ll /home/bruce/code/llvm-backend/build/install/bin/../lib/kllvm//libutil.a /home/bruce/code/llvm-backend/build/install/bin/../lib/kllvm//libstrings.a /home/bruce/code/llvm-backend/build/install/bin/../lib/kllvm//libnumeric_strings.a /home/bruce/code/llvm-backend/build/install/bin/../lib/kllvm//libio.a /home/bruce/code/llvm-backend/build/install/bin/../lib/kllvm//libcollections.a /home/bruce/code/llvm-backend/build/install/bin/../lib/kllvm//libParser.a /home/bruce/code/llvm-backend/build/install/bin/../lib/kllvm//libAST.a /home/bruce/code/llvm-backend/build/install/bin/../lib/kllvm//libBinaryKore.a /home/bruce/code/llvm-backend/build/install/bin/../lib/kllvm//libKOREPrinter.a /home/bruce/code/llvm-backend/build/install/bin/../lib/kllvm//liballoc.a /home/bruce/code/llvm-backend/build/install/bin/../lib/kllvm//libcollect.a /home/bruce/code/llvm-backend/build/install/bin/../lib/kllvm//libmeta.a /home/bruce/code/llvm-backend/build/install/bin/../lib/kllvm//libjson.a -Wl,-u,sort_table -ltinfo -lgmp -lgmpxx -lmpfr -lpthread -ldl -lffi -ljemalloc -I /home/bruce/code/llvm-backend/build/install/bin/../include/kllvm --std=c++17 -fno-stack-protector -fuse-ld=lld
  |   time=0.073
  time=0.204
```

### CODE REVIEW

**Comment by radumereuta** [comment]
Does anything interesting show up for C, for example?

**Comment by Baltoli** [comment]
For the C semantics, we see this:
```console
$ llvm-kompile -v --profile /home/bruce/code/c-semantics/.build/dist/lib/kcc/llvm/c-kompiled/definition.kore /home/bruce/code/c-semantics/.build/dist/lib/kcc/llvm/c-kompiled/dt main -- -o /home/bruce/code/c-semantics/.build/dist/lib/kcc/llvm/c-kompiled/interpreter -g -c -o /home/bruce/code/c-semantics/.build/dist/lib/kcc/llvm/c-kompiled/interpreter.o
+ /home/bruce/code/k/k-distribution/target/release/k/bin/llvm-kompile-codegen /home/bruce/code/c-semantics/.build/dist/lib/kcc/llvm/c-kompiled/definition.kore /home/bruce/code/c-semantics/.build/dist/lib/kcc/llvm/c-kompiled/dt/dt.yaml /home/bruce/code/c-semantics/.build/dist/lib/kcc/llvm/c-kompiled/dt 1
  time=7.377
+ /usr/lib/llvm-14/bin/opt -mem2reg -tailcallelim -tailcallopt tmp.5MS01jPavU -o tmp.HQo9h7wweX
  time=8.299
+ /home/bruce/code/k/k-distribution/target/release/k/bin/llvm-kompile-clang tmp.HQo9h7wweX main nolto -fno-stack-protector -v --profile -fuse-ld=lld --python-output-dir /home/bruce/code/c-semantics/.build/dist/lib/kcc/llvm/c-kompiled -o /home/bruce/code/c-semantics/.build/dist/lib/kcc/llvm/c-kompiled/interpreter -g -c -o /home/bruce/code/c-semantics/.build/dist/lib/kcc/llvm/c-kompiled/interpreter.o
  | + /usr/lib/llvm-14/bin/llc -tailcallopt tmp.HQo9h7wweX -mtriple=x86_64-pc-linux-gnu -filetype=obj -O0 --relocation-model=pic -o /home/bruce/code/c-semantics/.build/dist/lib/kcc/llvm/c-kompiled/interpreter.o
  |   time=20.702
  time=20.707
```

File: bin/llvm-kompile [hunk 1]
```diff
@@ -92,6 +90,11 @@ while [[ $# -gt 0 ]]; do
       kompile_clang_flags+=("-v")
       shift
       ;;
+    --profile)
```
Comment (Robertorosmaninho) [inline bin/llvm-kompile:93]: Shouldn't we print some error/warning when `--profile` is used without verbose?

Comment (Baltoli) [reply bin/llvm-kompile:93]: Sure, can do

**Review by Robertorosmaninho — APPROVED** [review]
LGTM! Thanks @Baltoli!

**Pull request merged**